src/plugins/event-plugins/JsErrorPlugin.ts

```diff
@@ -62,10 +65,15 @@ export class JsErrorPlugin implements Plugin {
     }
 
     private eventHandler = (errorEvent: ErrorEvent) => {
-        this.recordEvent(
-            JS_ERROR_EVENT_TYPE,
-            errorEventToJsErrorEvent(errorEvent, this.config.stackTraceLength)
-        );
+        if (!this.config.shouldIgnoreError(errorEvent)) {
```

Comment:
Isn't 'shouldIgnoreError' optional?

Reply:
So I have to admit, I had a wrong understanding of PartialConfigs and default values.. Now it makes sense that (1) default values should exist for non-optional parameters or else the code will break, (2) if a parameter is optional, the code should not rely on it.